### DISCUSSION TITLE
Does Hydro support conditional rendering?

### DISCUSSION THREAD

**Discussion by hmanbr** [hydrostack/hydro#107] [Q&A] [answered]
I'm looking for a way to do conditional rendering similar to Angular. If it's possible can someone give me an example? Thanks
```
//Angular code, if isEditing is true it will render #editTemplate
<div>
  <ng-container *ngIf="!isEditing; else editTemplate">
    <p>{{ text }}</p>
    <button (click)="editText()">Edit</button>
  </ng-container>

  <ng-template #editTemplate>
    <input [(ngModel)]="text" type="text">
    <button (click)="saveText()">Save</button>
  </ng-template>
</div>
```

**Comment by GiuseppeSpathis** [comment]
I think hydro is based on React, this is the code of conditional rendering in react

import React, { useState } from 'react';

const TextEditor = () => {
  const [isEditing, setIsEditing] = useState(false);
  const [text, setText] = useState('Your initial text');

  const editText = () => {
    setIsEditing(true);
  };

  const saveText = () => {
    setIsEditing(false);
  };

  return (
    <div>
      {/* If isEditing is false, show the text and the edit button, else show the input field and save button */}
      {!isEditing ? (
        <>
          <p>{text}</p>
          <button onClick={editText}>Edit</button>
        </>
      ) : (
        <>
          <input
            type="text"
            value={text}
            onChange={(e) => setText(e.target.value)}
          />
          <button onClick={saveText}>Save</button>
        </>
      )}
    </div>
  );
};

export default TextEditor;

**Comment by honkhonk-uk** [comment]
Hi all, please have a read at this:
https://usehydro.dev/introduction/motivation.html

**Comment by kjeske** [comment, marked as answer]
@hmanbr Hydro uses Razor syntax, so you can use `if` statement. [Here](https://learn.microsoft.com/en-us/aspnet/core/mvc/views/razor?view=aspnetcore-8.0#conditionals-if-else-if-else-and-switch) you can read more about that.
